MobileWallet/Screens/Home/HomeViewController.swift

```diff
@@ -92,7 +90,9 @@ class HomeViewController: UIViewController, FloatingPanelControllerDelegate, Tra
         super.viewDidLoad()
 
         self.refreshBalance()
-        tempRefreshTimer = Timer.scheduledTimer(timeInterval: 2, target: self, selector: #selector(self.refreshBalance), userInfo: nil, repeats: true)
+        TariEventBus.onMainThread(self, eventType: .balanceUpdate) { (_) in
+            self.refreshBalance()
```

Comment:
self inside a closure. we need to be sure we do not create a retain cycle.
`TariEventBus.onMainThread(self, eventType: .balanceUpdate) { [weak self] (_) in
guard let self = self else { return }
            self.refreshBalance()
 `